pkg/arvo/ted/get-gcp-jwt.hoon

```diff
@@ -0,0 +1,125 @@
+::  Thread that gets a JWT for use with Google Storage.
+::
+::  This thread expects settings-store to contain relevant fields from a GCP
+::  service account JSON file, e.g. as poked by urbit/sh/poke-gcp-account-json.
+::  Specifically, it depends on the `token_uri`, `client_email`,
+::  `private_key_id`, and `private_key` fields. If these fields are not in
+::  settings-store at the time the thread is run, it will fail.
+::
+::  The thread works by first constructing a self-signed JWT using the fields
+::  in settings-store. Then, it sends this JWT to the specified token URI
+::  (usually https://oauth2.googleapis.com/token), which gives us a JWT signed
+::  by Google. This token can then be used as a bearer token for requests to
+::  Google Storage.
+::
+::  The returned token has an expiration time of 60 minutes after the time at
+::  which the thread was called.
+::
+::
+/-  spider, settings
+/+  jose, pkcs, strandio
+=,  strand=strand:spider
+=,  format
+=,  jose
+=,  pkcs
+^-  thread:spider
+|=  arg=vase
+=/  m  (strand ,vase)
+^-  form:m
+;<  =bowl:spider  bind:m  get-bowl:strandio
+=>
+  |%
+  ::
+  ++  read-setting
+    |*  key=term
+    =/  m  (strand @t)  ^-  form:m
+    ;<  has=?  bind:m
+      (scry:strandio ? /gx/settings-store/has-entry/gcp-store/(same key)/noun)
```

Comment:
A little unusual to call `same` to insert a variable into a path with no modifications, normally we would just use:
`/gcp-store/[key]/noun`. Otherwise this looks fine.

Reply:
Sweet, fixed.